mem0/memory/main.py

```diff
@@ -667,7 +669,7 @@ def _create_memory(self, data, existing_embeddings, metadata=None):
         capture_event("mem0._create_memory", self, {"memory_id": memory_id, "sync_type": "sync"})
         return memory_id
 
-    def _create_procedural_memory(self, messages, metadata=None, prompt=None):
+    def _create_procedural_memory(self, messages, metadata: dict | None = None, prompt=None):
```

Comment:
Change this to `Optional[dict]` because this project supports Python >=3.9 and `|` was introduced in Python 3.10